macros/src/lib.rs

```diff
@@ -350,13 +366,12 @@ fn generate_into_root_node(
 
                 parse_args.push(quote! {
                     let #ident = <<#ty as lieutenant::ArgumentKind<#ctx_param>>::Parser
-                    as lieutenant::ArgumentParser<#ctx_param>>::default().parse(#ctx_ident,
-                    args[#i]).unwrap();
+                    as lieutenant::ArgumentParser<#ctx_param>>::default().parse(#ctx_ident, &mut input).unwrap();
                 });
 
                 i += 1;
-            }
-            _ => (),
+            },
+            Argument::Literal { .. } => parse_args.push(quote! { input.head(" "); }),
```

Comment:
Should we double-check that the literal is correct?

Reply:
Not necessary, since it's already checked by the dispatcher.